x-pack/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/upgrade_review_algorithms.md

```diff
@@ -0,0 +1,239 @@
+- [Scenarios](#scenarios)
+
+  - [Rule field doesn't have an update and has no custom value - `AAA`](#rule-field-doesnt-have-an-update-and-has-no-custom-value---aaa)
+  - [Rule field doesn't have an update but has a custom value - `ABA`](#rule-field-doesnt-have-an-update-but-has-a-custom-value---aba)
+  - [Rule field has an update and doesn't have a custom value - `AAB`](#rule-field-has-an-update-and-doesnt-have-a-custom-value---aab)
+  - [Rule field has an update and a custom value that are the same - `ABB`](#rule-field-has-an-update-and-a-custom-value-that-are-the-same---abb)
+  - [Rule field has an update and a custom value that are NOT the same - `ABC`](#rule-field-has-an-update-and-a-custom-value-that-are-not-the-same---abc)
+  - [Rule field has an update and a custom value that are the same and the rule base version doesn't exist - `-AA`](#rule-field-has-an-update-and-a-custom-value-that-are-the-same-and-the-rule-base-version-doesnt-exist----aa)
+  - [Rule field has an update and a custom value that are NOT the same and the rule base version doesn't exist - `-BC`](#rule-field-has-an-update-and-a-custom-value-that-are-not-the-same-and-the-rule-base-version-doesnt-exist----bc)
+
+## Useful information
+
+### Tickets
+
+- [Users can customize prebuilt detection rules](https://github.com/elastic/kibana/issues/174168) epic
+- [Implement single-line string diff algorithm](https://github.com/elastic/kibana/issues/180158)
+- [Implement number diff algorithm](https://github.com/elastic/kibana/issues/180160)
+- [Implement array of scalar values diff algorithm](https://github.com/elastic/kibana/issues/180162)
+
+### Terminology
+
+- **Base version**: Also labeled as `base_version`. This is the version of a rule authored by Elastic and unchanged by a user. The algorithms use it as a point of reference to see if the rule is being updated at all in addition to seeing if the user has customized anything themselves
+
+- **Current version**: Also labeled as `current_version`. This is the version of the rule that the user currently has installed. If any modifications were made to the "out of the box" `base_version` of the rule they would be represented by the `current_version` of the rule.
+
+- **Target version**: Also labeled as `target_version`. This is the version of the rule that contains the update from Elastic.
+
+- **Merged version**: Also labeled as `merged_version`. This is the version of the rule that we determine via the various algorithms. It could contain a mix of all the rule versions on a per-field basis to create a singluar version of the rule containing all relevant updates and user changes to display to the user.
+
+### Assumptions
+
+- All scenarios will contain at least 1 prebuilt rule installed in Kibana.
+- A new version will be available for rule(s).
+
+## Scenarios
+
+### Rule field doesn't have an update and has no custom value - `AAA`
+
+#### **Scenario: Rule field is any type**
```

Comment:
Trying to find a better name for all these scenarios that would be unique enough to use in a TOC, any suggestions?

Reply:
While not perfect, I think adding a three-letter type should work.

```markdown
### Rule field doesn't have an update and has no custom value - `AAA`
#### **Scenario: AAA - Rule field is any type**
```

Yep, some duplication, but it's not too bad IMO.